oauthlib/oauth2/rfc6749/request_validator.py

```diff
@@ -344,6 +344,31 @@ def get_id_token(self, token, token_handler, request):
         # the request.scope should be used by the get_id_token() method to determine which claims to include in the resulting id_token
         raise NotImplementedError('Subclasses must implement this method.')
 
+    def validate_id_token(self, token, scopes, request):
```

Comment:
I still wouldn't tie the naming here to ID Tokens specifically, as I wouldn't expect *all* JWT-based access tokens to be ID Tokens. Some servers might hand out special purpose JWT access tokens, next to their ID Tokens.

In that light, could this be named `validate_jwt_bearer_token` instead and have a companion `get_jwt_bearer_token` to create it?

Then people *could* alias those back to `get_id_token` if they want, but also have the option to split them up.

Reply:
This effort is to add OpenID connect support and for this use case the name is correct. Add JWT token support to OAuth is a different thing and can be handled in near future and at this point all methods that handle ID token and JWT token can converge but think it's a future refactoring not part of this PR.

Reply:
Does OIDC specify somewhere how to use ID Tokens as JWT Bearer tokens?

To me it seems that you *have* in fact implemented generic (OAuth, not OIDC) JWT Bearer token support (as also the generic class name `JWTToken` would suggest), except for the token validator functions that have ID Token specific names.

Generalising this later would likely lead to an inconvenient upgrade path, since then `JWTToken` is already expected to call the OIDC-specific validators.

Unless I'm missing the point of the PR, I see no reason to keep this OIDC-specific detail in. Or is there?

Reply:
@skion The OIDC don't say only:

* http://openid.net/specs/openid-connect-core-1_0.html#TokenResponse
* http://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthResponse
* https://tools.ietf.org/html/rfc6750

Make sense rename `validate_id_token` to `validate_jwt_bearer_token`